cmake --build . fails

I am using Ubuntu 18.04.

After cloning the fast-marching-method repository (HTTPS) to 
~/fmm-git/fast_marching_method

I made a build directory
~fmm-git/fmm_build

and moved there. Then I executed

cmake ../fast_marching_method/test -DCMAKE_BUILD_TYPE=Release

That seemed to go well, as the build files were written. But

cmake  --build . 

Failed with pages of errors. 

Can you provide any hints?

Thanks,
Tim

This error appears to be the non-trivial one:
```
fast_marching_method.hpp: In constructor 'thinks::fast_marching_method::detail::VaryingSpeedEikonalSolverBase<T, N>::VaryingSpeedEikonalSolverBase(const std::array<_Tp, _Nm>&, const std::array<long unsigned int, N>&, const std::vector<_Tp>&)':
fast_marching_method.hpp:2027:7: error: class 'thinks::fast_marching_method::detail::VaryingSpeedEikonalSolverBase<T, N>' does not have any field named 'EikonalSolverBase'
     : EikonalSolverBase(grid_spacing)
       ^~~~~~~~~~~~~~~~~
```
I was able to avoid the rest by enabling `-fpermissive`.

--- a/README.md
+++ b/README.md
@@ -15,12 +15,12 @@ Most of the functions in the single [header file](https://github.com/thinks/fast
 
 In the figure above, the green circle (_left_) was used as input to compute arrival times on a grid (_right_). Locations outside the circle have positive arrival times (or distances depending on interpretation), here shown in red. Similarly, locations inside the circle have negative arrival times, here shown in blue. The intensity gives the distance to the interface (i.e. circle boundary). This is why cells close to the interface appear black, since the red or blue component is small there. Next, we give an example demonstrating how to write code to generate an image similar to the one shown above.
 
-The input to the `ArrivalTime` function is given as grid cells with known distances (or arrival times depending on interpretation). The following code snippet computes a low resolution version of the image shown above.
+The input to the `ArrivalTime` function is given as grid cells with known distances (or arrival times depending on interpretation). The following code snippet (full working code [here](https://github.com/thinks/fast-marching-method/blob/master/test/minimal_example.cpp) ) computes a low resolution version of the image shown above.
 
 ```cpp
-using namespace std;
 namespace fmm = thinks::fast_marching_method;
 
+// Select points close to a true circle
 auto circle_boundary_indices = vector<array<int32_t, 2>>{
   {{5, 3}}, {{6, 3}}, {{7, 3}}, {{8, 3}}, {{9, 3}}, {{10, 3}}, {{4, 4}},
   {{5, 4}}, {{10, 4}}, {{11, 4}}, {{3, 5}}, {{4, 5}}, {{11, 5}}, {{12, 5}},
@@ -29,6 +29,8 @@ auto circle_boundary_indices = vector<array<int32_t, 2>>{
   {{5, 11}}, {{10, 11}}, {{11, 11}}, {{5, 12}}, {{6, 12}}, {{7, 12}},
   {{8, 12}}, {{9, 12}}, {{10, 12}},
 };
+
+// Specify distances of such points to the true circle
 auto circle_boundary_distances = vector<float>{
   0.0417385f, 0.0164635f, 0.0029808f, 0.0029808f, 0.0164635f, 0.0417385f,
   0.0293592f, -0.0111773f, -0.0111773f, 0.0293592f, 0.0417385f, -0.0111773f,
@@ -45,7 +47,7 @@ auto uniform_speed = 1.f;
 auto arrival_times = fmm::SignedArrivalTime(
   grid_size,
   circle_boundary_indices,
-  circle_boundary_times,
+  circle_boundary_distances,
   fmm::UniformSpeedEikonalSolver<float, 2>(grid_spacing, uniform_speed));
 ```
 

--- a/test/minimal_example.cpp
+++ b/test/minimal_example.cpp
@@ -65,15 +65,15 @@ int main([[maybe_unused]] int argc, [[maybe_unused]] char* argv[]) {
   auto initial_distances = std::vector<float>(grid_size[0] * grid_size[1], nan);
   for (std::size_t i = 0; i < num_seeds; ++i) {
     auto idx = circle_boundary_indices[i];
-    initial_distances[idx[0] * grid_size[1] + idx[1]] =
+    initial_distances[idx[0] + idx[1]* grid_size[0]] =
         circle_boundary_distances[i];
   }
 
   {
     std::cout << "Initial distance map (seeds):" << std::endl;
     std::size_t idx = 0;
-    for (std::size_t i = 0; i < grid_size[0]; ++i) {
-      for (std::size_t j = 0; j < grid_size[1]; ++j) {
+    for (std::size_t j = 0; j < grid_size[1]; ++j) {
+      for (std::size_t i = 0; i < grid_size[0]; ++i) {
         std::cout << std::setw(6) << std::fixed << std::setprecision(4)
                   << initial_distances[idx++] << '\t';
       }
@@ -89,8 +89,8 @@ int main([[maybe_unused]] int argc, [[maybe_unused]] char* argv[]) {
   {
     std::cout << "Distance map:" << std::endl;
     std::size_t idx = 0;
-    for (std::size_t i = 0; i < grid_size[0]; ++i) {
-      for (std::size_t j = 0; j < grid_size[1]; ++j) {
+    for (std::size_t j = 0; j < grid_size[1]; ++j) {
+      for (std::size_t i = 0; i < grid_size[0]; ++i) {
         std::cout << std::setw(6) << std::fixed << std::setprecision(4)
                   << arrival_times[idx++] << '\t';
       }

--- a/test/signed_arrival_time_test.cpp
+++ b/test/signed_arrival_time_test.cpp
@@ -600,8 +600,7 @@ TYPED_TEST(SignedArrivalTimeTest, Checkerboard) {
 
   auto const is_even = [](auto const i) { return i % 2 == 0; };
   auto const is_boundary = [=](auto const index) {
-    return std::all_of(std::begin(index), std::end(index), is_even) ||
-           std::none_of(std::begin(index), std::end(index), is_even);
+    return is_even( std::reduce(begin(index), end(index)) );
   };
 
   auto boundary_indices = std::vector<std::array<int32_t, kDimension>>();

